File: Week 3/Week3-2/app/src/main/java/com/example/week2_1/MainActivity.java
package com.example.week2_1;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.NumberPicker;

public class MainActivity extends AppCompatActivity {

    Button b1,b2,b3,b4;
    NumberPicker numberPicker;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.mymenu, menu);

        return super.onCreateOptionsMenu(menu);
    }

    @SuppressLint("NonConstantResourceId")
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.relative) {
            setContentView(R.layout.relative);

        } else if (id == R.id.linear) {
            setContentView(R.layout.linear);

        } else if (id == R.id.table) {
            setContentView(R.layout.table);

        } else if (id == R.id.constraint) {
            setContentView(R.layout.constraint);

        }
        return super.onOptionsItemSelected(item);
    }
}

File: Week 2/Week2-1/app/src/main/java/com/example/week1/MainActivity.kt
package com.example.week1

import android.os.Build
import android.os.Bundle
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.NumberPicker
import android.widget.TextView
import android.widget.Toast
import android.widget.Toolbar
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import com.example.week1.databinding.ActivityMainBinding
import com.google.android.material.textfield.TextInputEditText

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        val numPick = findViewById<NumberPicker>(R.id.numPick)
        numPick.minValue = 0
        numPick.maxValue = 1000
        val textView = findViewById<TextView>(R.id.textView3)
        val button = findViewById<Button>(R.id.button1)
        var amount = numPick.value
        val inputText = findViewById<EditText>(R.id.textInputEditText1)

        button.setOnClickListener {
            val field = inputText.text.toString().toIntOrNull()
            if (field == null) {
                amount = amount + numPick.value
                textView.text = "$$amount"
            } else {
                amount = amount + numPick.value + field!!
                textView.text = "$$amount"
            }


        }

    }



}